frontend/kubecloud/src/utils/api.ts

```diff
@@ -84,15 +86,15 @@ class ApiClient {
 
       // Add auth headers if required
       const requestHeaders: Record<string, string> = {
-        'Content-Type': 'application/json',
+        ...(contentType ? { 'Content-Type': contentType } : {}),
         ...(requiresAuth ? this.getAuthHeaders(customToken) : {}),
         ...headers
       }
 
       let response = await fetch(`${this.baseURL}${endpoint}`, {
         method,
         headers: requestHeaders,
-        body: body ? JSON.stringify(body) : undefined,
+        body: body instanceof FormData ? body : (body ? JSON.stringify(body) : undefined),
```

Comment:
```suggestion
        body: body instanceof FormData ? body : JSON.stringify(body),
```

Reply:
The body is optional; this was for handling that to avoid passing an undefined body to stringify